backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/impl/ElasticsearchBackendImpl.java

```diff
@@ -65,23 +67,22 @@
 	private final SearchBackendContext searchContext;
 
 	ElasticsearchBackendImpl(ElasticsearchClientImplementor client, GsonProvider gsonProvider, String name,
-			ElasticsearchWorkBuilderFactory workBuilderFactory,
+			ElasticsearchWorkBuilderFactory workFactory,
 			Gson userFacingGson,
 			ElasticsearchAnalysisDefinitionRegistry analysisDefinitionRegistry,
 			MultiTenancyStrategy multiTenancyStrategy) {
 		this.client = client;
 		this.name = name;
+		this.workProcessor = new ElasticsearchWorkProcessor( new LogErrorHandler(), client, gsonProvider, workFactory );
 		this.analysisDefinitionRegistry = analysisDefinitionRegistry;
 		this.multiTenancyStrategy = multiTenancyStrategy;
-		this.streamOrchestrator = new ElasticsearchStubWorkOrchestrator( client, gsonProvider );
-		this.queryOrchestrator = new ElasticsearchStubWorkOrchestrator( client, gsonProvider );
+		this.streamOrchestrator = workProcessor.createStreamOrchestrator( name );
+		this.queryOrchestrator = workProcessor.createStreamOrchestrator( name );
```

Comment:
Good idea, Moreover, I would like to remove `streamOrchestrator` instance from `IndexingBackendContext`, so that it will be used just by initializeIndex method. Do you agree with that idea?